app/lib/CustomHttpErrorHandler.scala

```diff
@@ -16,17 +19,26 @@ class CustomHttpErrorHandler(
     env: Environment,
     config: Configuration,
     sourceMapper: Option[SourceMapper],
-    router: => Option[Router]
+    router: => Option[Router],
+    val assets: AssetsResolver
 )(implicit val ec: ExecutionContext) extends DefaultHttpErrorHandler(env, config, sourceMapper, router) with LazyLogging {
 
+  implicit val ar = assets
+
   override def onClientError(request: RequestHeader, statusCode: Int, message: String = ""): Future[Result] =
     super.onClientError(request, statusCode, message).map(_.withHeaders(CacheControl.defaultCacheHeaders(30.seconds, 30.seconds): _*))
 
   override protected def onNotFound(request: RequestHeader, message: String): Future[Result] =
-    super.onNotFound(request, message).map(_.withHeaders(CacheControl.defaultCacheHeaders(30.seconds, 30.seconds): _*))
+    Future.successful(
+      NotFound(views.html.react("Error 404", "error-404-page", "error404Page.js")(ar, request))
```

Comment:
I would probably `import views.html.react` at the top of the file and just use `react` here